### PR TITLE
Default publishing to v3 and more cleanup

### PR DESCRIPTION
**Pull request by mmitche**
Set the default publishing version to v3. This was previously done by setting this in PublishArtifactsInManifest.proj. However, this has no effect in this project now. Manifest generation is where it is critical to get the right value. This should happen in Publish.proj, PublishArtifactsInManifest, and GenerateBuildManifest. All should default to 3/Next unless specified.

This still allows for overriding publishing versions in eng/Publishing.props if it became required (e.g. v4 publishing). But makes it non-critical.

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Review by missymessa — DISMISSED** [review]
LGTM shipit

**Comment by mmitche** [comment]
Added a bunch more cleanup to remove legacy code.

**Comment by mmitche** [comment]
I'm going to run an official build to ensure that this doesn't break publishing.

**Comment by mmitche** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=2164207&view=results

**Comment by mmitche** [comment]
Publishing testing looks great.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj [hunk 1]
```diff
@@ -27,6 +27,9 @@
   <PropertyGroup>
     <PublishToSourceBuildStorage>false</PublishToSourceBuildStorage>
     <PublishToSourceBuildStorage Condition="'$(DotNetOutputBlobFeedDir)' != ''">true</PublishToSourceBuildStorage>
+
+    <!-- Default publishing infra target is 3. -->
+    <PublishingInfraVersion Condition="'$(PublishingInfraVersion)' == ''">3</PublishingInfraVersion>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj:32]: @mmitche shouldn't this be `PublishingVersion` instead?

**Pull request merged**